mantidproject/autoreduce
# ICAT Cache

Fixes #20, addresses #128 somewhat.

The class `ICATCache` mimics the behaviour of `ICATCommunication`, but keeps a local cache which it will check before using ICATCommunication to send any queries to ICAT. The cache lifetime is `CACHE_LIFETIME` in the settings, currently set to one hour.

In the event that ICAT can't be connected to, it will serve stale cache objects if it has them. This means that if an object has been queried for before, subsequent queries from the frontend will succeed when ICAT is unavailable.

To test:
* Check that all permissions, run/instrument/experiment information, etc., are as they should be on the frontend.
* Temporarily change the cache lifetime to something short and check that the cache is refreshed when it should be (cache objects have a `created` field).
* Temporarily break ICAT access somehow and check that the webapp continues working for objects that have been queried before.
